src/components/measureGroups/SupplementalElements.tsx

```diff
@@ -1,7 +1,294 @@
+        `Please complete the CQL Editor process before continuing`,
+        true
+      );
+    }
+  }, [measure]);
+
+  const submitForm = (values) => {
+    const submitMeasure = {
+      ...measure,
+    };
+
+    //to-do: update measure with new supplemental data elements
+    measureServiceApi
+      .updateMeasure(submitMeasure)
+      .then(() => {
+        handleToast(
+          "success",
+          `Supplement Data Element Information Saved Successfully`,
+          true
+        );
+        updateMeasure(submitMeasure);
+      })
+      .catch(() => {
+        const message = `Error updating measure "${measure.measureName}"`;
+        setErrorMessage(message);
+
+        handleToast(
+          "danger",
+          `Error updating measure "${measure.measureName}"`,
+          true
+        );
+      });
+  };
+
+  useEffect(() => {
+    const subscription = measureStore.subscribe(setMeasure);
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, []);
+
+  // Route handle store will give warning message
+  // if form is dirty and user ties to navigate across SPAs
+  const { updateRouteHandlerState } = routeHandlerStore;
+  const [discardDialogOpen, setDiscardDialogOpen] = useState(false);
+
+  useEffect(() => {
+    updateRouteHandlerState({
+      canTravel: !formik.dirty,
+      pendingRoute: "",
+    });
+  }, [formik.dirty]);
+
+  const handleDesciptionChange = (field, value, index) => {
+    selectedSupplementalDataList.filter((selected) => {
+      if (selected === field) {
+        formik.setFieldValue(field, value);
+      }
+    });
+  };
+
+  return (
+    <form
+      id="measure-details-form"
+      onSubmit={formik.handleSubmit}
+      data-testid="supplemental-data-form"
+    >
+      <div className="content">
+        <div className="subTitle">
+          <h2>Supplemental Data</h2>
+          <div className="required">
+            <Typography
+              style={{ fontSize: 14, fontWeight: 300, fontFamily: "Rubik" }}
+            >
+              <span style={asterisk}>*</span>
+              Indicates required field
+            </Typography>
+          </div>
+        </div>
+        <div>
+          {supplementalDataElementList && (
+            <>
+              <div tw="mb-4 w-1/2">
+                <div className="left-box">
+                  <AutoComplete
+                    multiple
+                    id="supplementalDataElements"
+                    data-testid="supplementalDataElements"
+                    label="Definition"
+                    placeholder=""
+                    required={false}
+                    disabled={!canEdit}
+                    error={
+                      formik.touched.supplementalDataElements &&
+                      formik.errors["supplementalDataElements"]
+                    }
+                    helperText={
+                      formik.touched.supplementalDataElements &&
+                      formik.errors["supplementalDataElements"]
```

Comment:
Are there any inline errors to display for AutoComplete component?

Reply:
@RohitKandimalla No there shouldn't be, I'll remove it